### ISSUE TITLE
Limit actions that trigger undo tree updating 

### ISSUE THREAD

**Issue by davidklsn** [mbbill/undotree#75]
What I want is to stop undoTree from updating when I am leaving insert mode. I have tried this in my .vimrc by removing 'InsertLeave'. 

let s:auEvents = "BufEnter,CursorMoved,BufWritePost"
augroup Undotree
    exec "au! ".s:auEvents." * call UndotreeUpdate()"
augroup END

But it wont work, and I have not figured out any other way to make it work.

Sorry if this is not an issue and just a case of my incompetence :)

**Comment by mbbill** [comment]
Thanks for the feedback! may I ask why do you need to disable the auto updating on normal mode? Is it conflict with some other plugins?
It is possible that the undo history changes when you're not in insert mode, and in that case the undotree will out of sync with the content.

**Comment by davidklsn** [comment]
The first reason is that vim is locking up very briefly each time i go from insert to normal mode when the undo tree is updating. Although I am not sure that it is actually locking up, but that is what it feels like :) 

The second reason is that the undo tree is a bit to specific for me when showing changes with only seconds apart, see the image:
<img width="271" alt="screen shot 2017-09-07 at 18 52 32" src="https://user-images.githubusercontent.com/1679182/30175148-d24db122-93fd-11e7-8204-37a5e05fd7bf.png">

It is just a bit to detailed for me. I usually jumps back to changes I made way further back in time than seconds.

So what I really would like is to have the undo tree to update only when I save the file or alternatively automatically less frequent.

But I just wanna say that I really enjoy the plugin and find it very useful!

**Comment by mbbill** [comment]
Now I understand, and thanks for the feedback! Yes, the undotree window will update its content whenever a new undo node is generated, like leaving insert mode for example. However, it's pretty much a VIM behaviour, and the plugin itself can't do too much about it.
Sometimes it is a bit too detailed, but that's how vim works. It's also one of the reason I added a timestamp on it because it's much useful to estimate a change based on its time rather than the serial numbers.
I'm afraid it's a little bit hard to force the undotree to out of sync with the content because it's designed to work in real-time. However, I never experienced a 'lock up' when I go from insert to normal, and I tried to make it as quick as possible (by using cache and some other techniques). Not sure if it's caused by some conflict with the other plugins.
